### ISSUE TITLE
[feat] As dev, I want to be able to inflate all the `flat_*.Rmd` files

### ISSUE THREAD

**Issue by statnmap** [ThinkR-open/fusen#96]
- [x] This issue needs the issue #194, #196, #198, #202 to be done first

## Objective

Inflate all the flat files listed in `config_fusen.yaml` and for which inflate-related parameters were specified.
We need to use the function `read_inflate_params()`

## Validation

- [x] The user is informed if inflate_all is run without config file
![image](https://user-images.githubusercontent.com/21193866/235078653-1c4efbc8-561e-4a5a-aadb-f6a2798d97b7.png)
- [x] The user is informed if a flat file exists and will not be inflated
![image](https://user-images.githubusercontent.com/21193866/235080549-cae4e52c-715c-4027-916a-131d1c1ee869.png)

- [x] There is an `inflate_all()` function which will inflate all the active flat_file present in `config_fusen.yaml` with the correct inflate parameters 
- [x] The user is informed of the list of flat files that will be inflated
- [x] The user is informed of the list of flat files that wont be inflated
- [x] These functions are documented and tested
- [x] There is a dedicated vignette which explains the process to inflate all the flat files.
- [x] There is no error with an old package that passed through `register_all_to_config()`, thus a section "keep" in the config file.
- [x] `inflate_all()` runs document and check only once.

## Technicals

-  [x] Write a function `inflate_all()`which:
  - [x] runs the function `read_inflate_params()` to create the list of inflate parameters
  - [x] uses a base R equivalent of purrr::walk to inflate each file (see https://purrr.tidyverse.org/articles/base.html)
for a given flat file it will look like : do.call(fusen::inflate(), my_list_of_parameters) 
  - [x]  account for `vignette_name = c("name" = "index")` format
 
- [x]  Complete the flat_file flat_read_inflate_params.Rmd for the future vignette (this should probably be renamed flat_inflate_all.Rmd)
See https://thinkr-open.github.io/fusen/articles/tips-and-tricks.html#how-to-inflate-multiple-flat-files

## Unit tests

Tests to be made : 

- [ ] `inflate_all()` when no config_fusen.yml exists -> error

- [ ]  start from a clean package created with {fusen}
  - [ ] create a flat file
  - [ ] inflate it individually -> config_fusen.yml contains the inflate parameters
  - [ ] add a new function in the flat file
  - [ ] run inflate_all 
  - [ ] check whether the new function exists in config_fusen.yml, R/, tests/ and man/

- [ ] start from a clean package created with {fusen}
  - [ ] create 2 flat files in a dev/ folder
  - [ ] used a predefined config_fusen.yml which looks like `system.file("inflate_all/config_fusen_with_inflate_parameters.yaml", package = "fusen")`. i.e in this file we have inflate parameters for one file and nothing for the other
  - [ ] after `inflate_all()` we must find the functions of only the flat file for which inflate parameters where stated in config_fusen.yml, in R/ and tests/ 

### Note:

Dealing with flat files present in config_fusen.yml without staded inflate-parameters, should be done in a future issue

### Vigilance : 

- check if we can account for vignette_name = c("name" = "index") format

- check the behaviour of `inflate_all()` when the following situation occurs : "I foresee a problem in using the "basename" of the pkg when we will want to run inflate_all(), depending on from where we run it, but I am sure we will detect it when we will really implement it... I guess, this PKG parameter may not be stored, and the user will have to define it in function inflate_all by themselves"

**Comment by statnmap** [comment]
As you propose some code, that's better with a new issue. @MargotBr

**Comment by ymansiaux** [comment]
I store here the content of the original issue.

This issue was designed at first with the hypothesis that we would read on the yaml of each flat file the inflate parameters.

With the work started on issue #194 and #196, the `inflate_all` operation will be done based on the content of the `config_fusen.yaml` file.


----

This issue needs the issue #191 to be done first

## Validation

- [ ] A function `get_inflate_params()` retrieves the inflate parameters stored in the YAML header of each flat, to register them in the "config_fusen.yml" file
- [x] In the vignette: https://thinkr-open.github.io/fusen/articles/tips-and-tricks.html#how-to-inflate-multiple-flat-files
- [ ] A function `inflate_all()` can inflate all flats that are not deprecated (not stored in "dev/dev_history/flat_*"
- [ ] The function detects if a flat file does not have any parameters in the "config_fusen.yml", or parameters in the YAML of the flat file to retrieve. And let the user know that it must inflate the concerned flat file manually or use the default arguments
- [ ] ~~The function uses all options if the "inflate" chunks.~~
- [ ] ~~The function tests if there is something else than "inflate()" in the inflate chunk and warn the user~~
- [ ] ~~`fusen::inflate()` without parameter inflates the current flat file (as today) but uses all options in the "inflate" chunk, as for `inflate_all()`~~

## Technicals

~~- Retrieve the full content of the "inflate" chunk to get the parameters. That may be easier inside the `inflate()` function itself because all params are there: See https://github.com/ThinkR-open/fusen/issues/190~~
~~- Think about a way to store the inflate options inside the "fusen_config.yaml" file, as with files list
    - Use a specific function `flat_params_to_config()`, even used to define the `state` already implemented~~
~~- Note  that I can run the `inflate()` without parameters~~

**Comment by ymansiaux** [comment]
History :

🔴 Closed issues:

https://github.com/ThinkR-open/fusen/issues/192 

In this issue, I tried to parse the content of the `inflate` chunk. We changed the way we want to reach our goal to have an `inflate_all`, but it could still be useful in a transition phase.


---
~~- [ ] Detect inflate options in the `inflate` chunk. Make the chunk named `inflate` mandatory~~
~~- [ ] Allow the inflate option to be listed in the YAML of the flat file~~
~~- [ ] Make sure it is consistent with the ones in the `dev-inflate` chunk~~

- [ ] Inform if previous {fusen} version inflated is not the same as current: Probably need to re-inflate all files for version <= 0.4.1
    - [x] Add fusen Version in DESCRIPTION when inflating
- [ ] Register inflate parameters, so that we can inflate all  
    + https://github.com/ThinkR-open/fusen/issues/96

**Comment by statnmap** [comment]
That should be good for this one now.
